source/s3_meta_request.c

```diff
@@ -2081,13 +2098,25 @@ struct aws_future_bool *aws_s3_meta_request_read_body(
         return aws_parallel_input_stream_read(meta_request->request_body_parallel_stream, offset, buffer);
     }
 
+    /* Further techniques are synchronous... */
+    struct aws_future_bool *synchronous_read_future = aws_future_bool_new(meta_request->allocator);
+
+    /* If using async-writes, call function which fills the buffer and/or hits EOF  */
+    if (meta_request->request_body_using_async_writes == true) {
+        bool eof = false;
+        if (s_s3_meta_request_read_from_pending_async_writes(meta_request, buffer, &eof) == AWS_OP_SUCCESS) {
```

Comment:
`s_s3_meta_request_read_from_pending_async_writes` doesn't seem to return anything other than AWS_OP_SUCCESS. Maybe we can make it void instead

Reply:
I wasn't sure whether to turn some of the asserts into actual errors ...
but then I hate when I see error-handling for things that shouldn't be possible, it makes me wonder if they really are possible and I should be handling stuff like that everywhere

let's live bold: I'll make them FATAL_ASSERTS